### PR TITLE
(feat) Add ability to toggle side menu in tablet mode

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

At the moment, users are unable to display the left-side-nav. This PR seeks to fix this issue.

## Screenshots
Kapture 2022-05-25 at 09 36 42

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by brandones** [comment]
Looks like you didn't quite wait for your gif to upload, @donaldkibet

File: packages/esm-outpatient-app/src/visit-header/visit-header-side-menu.component.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+interface VisitHeaderSideMenu {
+  isExpanded: boolean;
+  toggleSideMenu: () => void;
+}
+
+const VisitHeaderSideMenu: React.FC<VisitHeaderSideMenu> = ({ isExpanded, toggleSideMenu }) => {
+  const menuRef = useRef(null);
+
+  useEffect(() => {
```
Comment (brandones) [inline packages/esm-outpatient-app/src/visit-header/visit-header-side-menu.component.tsx:12]: Why not [useOnClickOutside](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-framework/docs/API.md#useonclickoutside)?

**Review by brandones — APPROVED** [review]

File: packages/esm-outpatient-app/src/visit-header/visit-header-side-menu.component.tsx (hunk 1 above)
Comment (donaldkibet) [reply packages/esm-outpatient-app/src/visit-header/visit-header-side-menu.component.tsx:12]: The `useOnClickOutside` hook handles clicking outside the side menu, we also want to close the side menu when the user clicks a link and popstate event happens.

Comment (brandones) [reply packages/esm-outpatient-app/src/visit-header/visit-header-side-menu.component.tsx:12]:
```ts
  const menuRef = useOnClickOutside(toggleSideMenu);

  React.useEffect(() => {
    window.addEventListener('popstate', toggleSideMenu);
    return window.addEventListener('popstate', toggleSideMenu);
  }, [toggleSideMenu]);
```

?

**Comment by brandones** [comment]
@donaldkibet Could you upload the gif when you get the chance?

File: packages/esm-outpatient-app/src/visit-header/visit-header.scss [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 .topNavHeader {
   top: var(--omrs-offline-banner-height);
-  @include brand-02(background-color);
+  @include brand-01(background-color);
```
Comment (hadijahkyampeire) [inline packages/esm-outpatient-app/src/visit-header/visit-header.scss:7]: @donaldkibet why did you change this back to brand-01, it is supposed to be brand-02 ```#004144``` according to mockups and what Ciaran mentioned

Comment (donaldkibet) [reply packages/esm-outpatient-app/src/visit-header/visit-header.scss:7]: No, in particular, we can correct it

Comment (hadijahkyampeire) [reply packages/esm-outpatient-app/src/visit-header/visit-header.scss:7]: Okay, great, I will put it back to brand-02, it needs to be different from the normal NavBar.
